File: examples/python/lambda_function.py
from __future__ import print_function
import os
import sys
import subprocess

# Just a test lambda, run with:
# docker run --rm -v "$PWD":/var/task lambci/lambda:python2.7
# OR
# docker run --rm -v "$PWD":/var/task lambci/lambda:python3.6
# OR
# docker run --rm -v "$PWD":/var/task lambci/lambda:python3.7 lambda_function.lambda_handler

def lambda_handler(event, context):
    context.log('Hello!')
    context.log('Hmmm, does not add newlines in 3.7?')
    context.log('\n')

    print(sys.executable)
    print(sys.argv)
    print(os.getcwd())
    print(__file__)
    print(os.environ)
    print(context.__dict__)

    return {
        "executable": str(sys.executable),
        "sys.argv": str(sys.argv),
        "os.getcwd": str(os.getcwd()),
        "__file__": str(__file__),
        "os.environ": str(os.environ),
        "context.__dict__": str(context.__dict__),
        "ps aux": str(subprocess.check_output(['ps', 'aux'])),
        "event": event
    }

File: examples/ruby/lambda_function.rb
require 'pp'

# docker run --rm -v "$PWD":/var/task lambci/lambda:ruby2.5 lambda_function.lambda_handler

def lambda_handler(event:, context:)
  info = {
    'event' => event,
    'ENV' => ENV.to_hash,
    'context' => context.instance_variables.each_with_object({}) { |k, h| h[k] = context.instance_variable_get k },
    'ps aux' => `ps aux`,
    'proc environ' => `xargs -n 1 -0 < /proc/1/environ`,
  }

  pp info

  return info
end
